Test/TestFramework/Shared/AssertExtensions.cs

```diff
@@ -0,0 +1,252 @@
+﻿namespace Microsoft.ApplicationInsights.TestFramework
+{
+    using System;
+    using System.Diagnostics.CodeAnalysis;
+    using System.Reflection;
+    using System.Threading.Tasks;
+    using Microsoft.VisualStudio.TestTools.UnitTesting;
+    using System.Collections;
+    using System.Collections.Generic;
+
+#if !NET40
+    using TaskEx = System.Threading.Tasks.Task;
+#endif
+
+
+    internal class AssertEx
+    {
+        public static void AreEqual<T>(IEnumerable<T> expected, IEnumerable<T> actual)
+        {
+            var a = actual.GetEnumerator();
+
+            foreach(var t in expected)
```

Comment:
This seems to rely on equal ordering of the collections. I see that this assert is mostly used for Arrays above, so should not be an issue right now, but someone might want to use it to compare non-ordered collections later and this limitation is not implied in the method name.